File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLOUR_TO_VALUE = {
    "black" => 0,
    "brown" => 1,
    "red" => 2,
    "orange" => 3,
    "yellow" => 4,
    "green" => 5,
    "blue" => 6,
    "violet" => 7,
    "grey" => 8,
    "white" => 9
  }

  def self.value(colours)
    colours.map(&COLOUR_TO_VALUE).join("").to_i
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
export const colorCode = (code) => {
  return COLORS.indexOf(code);
};

export const decodedValue = (colours) => {
  let num = 0;

  for (let i = 0; i < 2; i += 1) {
    if (i === 0) {
      num += colorCode(colours[i]) * 10;
    } else {
      num += colorCode(colours[i]);
    }
  }

  return num;
};

const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];
